### DISCUSSION TITLE
Alacritty and Firefox Not Running in labwc

### DISCUSSION THREAD

**Discussion by CloisteredNeuron** [labwc/labwc#1879] [General]
Any ideas on what might be amiss here?

Probably not related but I am also having performance issues described in https://github.com/labwc/labwc/discussions/1874.

I'm running labwc in a VirtualBox VM.  In https://i.postimg.cc/RFnfkPyC/labwc-config.gif I show how the VM is configured, that firefox and alacritty are installed, how menu.xml is configured to run alacritty and firefox, and how I start labwc.

In https://i.postimg.cc/MTW09Hyn/labwc-alacritty.gif I show how I run alacritty from the foot terminal, how it generates errors and does not run, and the messages I see in the console after exiting labwc.

In https://i.postimg.cc/B6Mzgd1G/labwc-firefox.gif I show how I run firefox from the foot terminal, how it terminates labwc and displays messages in the console.  This image specifically mentions wayland in the errors.

**Comment by Consolatis** [comment]
If you disable the 3d acceleration in virtualbox, does the same thing happen?

**Comment by CloisteredNeuron** [comment]
Actually labwc will not start with no 3d acceleration.  See attached image for errors:

https://i.postimg.cc/yNcFckPY/labwc-No3-DAcceleration.gif

**Comment by CloisteredNeuron** [comment]
To get Firefox to start I set MOZ_ENABLE_WAYLAND=1 in the environment file. While Firefox did start it was slow and disabling Firefox acceleration extended the usefulness to allowing me to search for taco. Total time to search on taco was about 2 minutes and I did get a page of results but then Firefox became unresponsive.

https://i.postimg.cc/26GDz1GX/labwc-firefox-safe-mode.gif

There seems to be some disconnect with wayland and VirtualBox.

**Comment by droc12345** [comment]
I'm not sure virtualbox runs wayland, at least natively. I think it needs xwayland.
And I'm seeiing messages about drm, I'm not sure that's a valid backend for virtualbox.

google wayland and virtualbox for more

Edit to add: LInux does have a graphics driver for virtualbox
DRM_VMWGFX
Choose this option if you would like to run 3D acceleration in a VMware virtual machine.

**Comment by CloisteredNeuron** [comment]
HI droc12345,

Thanks for the suggestion.  I found an Oracle employee state: "VirtualBox 7.0.x will work only with Xwayland."
https://forums.virtualbox.org/viewtopic.php?p=547412&sid=525db9ae2a3320d4a3402d0ee9c68625#p547412

Just to make sure this was still the case I grabbed the latest version of VirtualBox and I am still getting the same response.

I installed Gnome from Debian and ran it in Gnome Classic Wayland mode and it runs without issue so I looked for Xwayland.  While I did not find Xwayland I did find /usr/libexec/gdm-wayland-session gnome-session-classic in HTOP.  

DRM_VMWGFX sounds like an interesting option.  Do you know how to install it in labwc?

Also do you know if labwc has been tested using any virtualization software on Windows?  Maybe using a different virtualization software is the answer.

A big concern I had with Gnome was is it really running wayland.  In the image you can see xinput capturing keypresses from a calculator and userid/pw field from a browser.  https://i.postimg.cc/fTkdvhS6/xinput-sees-all-input.gif

That image was from Gnome classic in xorg mode.  In wayland mode xinput captured nothing so that is a bit of a sanity check.  It is so cool labwc is being developed.   There are not a lot of light weight desktop options that prevent your calculator/notepad/browser from capturing your im's, emails, sudo pws, or financial account credentials.

Thanks!

**Comment by CloisteredNeuron** [comment]
In addition to Gnome, I also tried running Fedora-Sway-Live-x86_64-40-1.14 (fedoraproject.org/spins/sway/) in VirtualBox.  It too is based upon wayland.  It starts up but locks up when I try to install it.  So, the issue is not just with labwc.  Perhaps you could share notes with the Fedora group on why this might be occurring.  Lots of people get hands on experience with an OS using VirtualBox so it would be good if labwc and sway could work in VirtualBox.  I believe Fedora does have a Gnome group and Gnome wayland from Debian at least does run well in VirtualBox.  Perhaps the Fedora Gnome group could provide info what steps they took to get Gnome to work in VirtualBox.
